library/alloc/src/vec/mod.rs

```diff
@@ -2141,7 +2141,9 @@ impl<T, A: Allocator> Vec<T, A> {
     unsafe fn split_at_spare_mut_with_len(
         &mut self,
     ) -> (&mut [T], &mut [MaybeUninit<T>], &mut usize) {
-        let Range { start: ptr, end: spare_ptr } = self.as_mut_ptr_range();
+        let ptr = self.as_mut_ptr();
+        // SAFETY: see slice::as_ptr_range() for why `add` here is safe.
```

Comment:
This is safe because we always know that our buffer is at least `self.len` elements big.